### DISCUSSION TITLE
Error in Symfony files while making module

### DISCUSSION THREAD

**Discussion by WaqarAnwar** [symfony/symfony#45454] [Q&A]
I have been following symfony tutorial and i am here: 
https://symfony.com/legacy/doc/jobeet/1_4/en/03?orm=Doctrine

Problem is with the code: 
`php symfony doctrine:generate-module --with-show --non-verbose-templates frontend job JobeetJob`

When i run the webpage, it shows error: Parse error: syntax error, unexpected 'Deprecated' (T_STRING), expecting ',' or ';' in C:\dev\sfprojects\jobeet\apps\frontend\modules\job\templates\indexSuccess.php on line 33

And here is the line 33 and its surrounding code:

``````
<td><a href="<?php echo url_for('job/show?
Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359

Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359
id='.$jobeet_job->getId()) ?>"><?php echo $jobeet_job->get
----------->Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359

Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359
Id() ?></a></td>
``````
Same type of error is in another file showsuccess.php:
``````
<td><?php echo $jobeet_job->get
------->Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359

Deprecated: preg_replace(): The /e modifier is deprecated, use preg_replace_callback instead in C:\dev\sfprojects\jobeet\lib\vendor\symfony\lib\util\sfToolkit.class.php on line 359
Id() ?></td>
``````
kindly tell what is the issue and how can i resolve it? Thanks in advance.

**Comment by michaoj** [comment]
May I ask you why you're following a tutorial for the version 1 (which is legacy) while the version 6 is out?

By the way, the error you're facing is happening because the preg_replace function as changed. Therefore either don't use the doctrine:generate:module command and create the module "by hand".

Otherwise switch to symfony 6 :smile:

**Comment by michaoj** [comment]
In that case, your best choice would be to create module by hand by looking on how it's done in your existing project.

Another kinda hackish solution would be to look inside the file vendor\symfony\lib\util\sfToolkit.class.php on line 359 and fix the call by yourself but this update will be lost if you perform another composer install/update
